### ISSUE TITLE
Login using LinkedIn

### ISSUE THREAD

**Issue by jgrandja** [jgrandja/oauth2-samples#28]

**Comment by jgrandja** [comment]
@rwinch I configured a LinkedIn client (for testing) and based on their [OAuth implementation](https://developer.linkedin.com/docs/oauth2),  the Client Credentials are set as URL params instead of `Authorization` header when exchanging the authorization code for an access token. See the section **Step 3 — Exchange Authorization Code for an Access Token** . So basically the client secret is exposed which is not ideal.

Also, in the same section, the Access Token response only has the attributes `access_token` and `expires_in`. It does not have `token_type` which is required according to the spec.

So basically, LinkedIn does not work with the current codebase as they have not implemented to spec.
I'm not going to make any changes to the current codebase to make it work until we have a discussion on this.

**Comment by jgrandja** [comment]
Added support for client authentication using form post via #77 so this resolves the 1st issue but the 2nd issue is still outstanding (missing `token_type` attribute on token response).
